IdentityCore/tests/util/MSIDTestCacheDataSource.m

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDTestCacheDataSource.h"
+#import "MSIDTokenCacheKey.h"
+#import "MSIDTokenSerializer.h"
+#import "MSIDKeyedArchiverSerializer.h"
+
+@interface MSIDTestCacheDataSource()
+{
+    NSMutableDictionary<NSString *, NSData *> *_cacheContents;
+    NSDictionary *_wipeInfo;
+}
+
+@end
+
+@implementation MSIDTestCacheDataSource
+
+#pragma mark - Init
+
+- (instancetype)init
+{
+    self = [super init];
+    
+    if (self)
+    {
+        _cacheContents = [NSMutableDictionary dictionary];
+    }
+    
+    return self;
+}
+
+#pragma mark - MSIDTokenCacheDataSource
+
+- (BOOL)setItem:(MSIDToken *)item
+            key:(MSIDTokenCacheKey *)key
+     serializer:(id<MSIDTokenSerializer>)serializer
+        context:(id<MSIDRequestContext>)context
+          error:(NSError **)error
+{
+    if (!item
+        || !key
+        || !serializer)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Invalid or missing parameter", nil, nil, nil, nil, nil);
```

Comment:
I am still kind of torn in having this as an internal vs invalid parameter.
I was on internal before, more and more I look into this, it might be better classified as an invalid parameter.

Reply:
We could also actually infer from the description but i think it be clearer if we set it invalid param.

Reply:
fixed